src/engines/engine-decorator.js

```diff
@@ -48,6 +50,15 @@ class EngineDecorator extends FakeEventTarget implements IEngineDecorator {
     return activeDecorator ? activeDecorator.dispatchEvent && activeDecorator.dispatchEvent(event) : super.dispatchEvent(event);
   }
 
+  _reset(): void {
+    this._eventManager.removeAll();
```

Comment:
same here, do we event reset the proxy and re-register to the events? 
Also, the dependency is backward here, player creates this instance so need to call from player on destroy of proxy and not listen to events.